src/zope/file/download.py

```diff
@@ -56,8 +56,12 @@ def getHeaders(context, contentType=None, downloadName=None,
         downloadName = downloadName or context.__name__
         if contentDisposition:
             if downloadName:
+                # Headers must be native strings. Under Py2, we probably
+                # need to encode the name
+                if str is bytes and not isinstance(downloadName, bytes):
+                    downloadName = downloadName.encode("utf-8")
                 contentDisposition += (
-                    '; filename="%s"' % downloadName.encode("utf-8")
+                    '; filename="%s"' % downloadName
```

Comment:
Alas, trying to test this smuggling reveals a bug in `zope.app.wsgi.testlayer:FakeResponse.getOutput()`: It assumes that all headers are unicode. Under Python 2, when it does `h.encode('latin-1')`, that first tries to decode the byte str `h` into unicode using the default encoding, ASCII, which fails.